cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -126,80 +132,83 @@ struct fixed_width_type_converter {
 };
 
 /**
- * @brief Creates a `device_buffer` containing the elements in the range `[begin,end)`.
- *
- * @tparam ElementTo The element type that is being created (non-`fixed_point`)
- * @tparam ElementFrom The element type used to create elements of type `ElementTo`
- * @tparam InputIterator Iterator type for `begin` and `end`
- * @param begin Beginning of the sequence of elements
- * @param end End of the sequence of elements
- * @return rmm::device_buffer Buffer containing all elements in the range `[begin,end)`
+ * @brief Convert fixed-point to Representation value.
  */
-template <typename ElementTo,
-          typename ElementFrom,
-          typename InputIterator,
-          typename std::enable_if_t<not cudf::is_fixed_point<ElementTo>()>* = nullptr>
-rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
-{
-  static_assert(cudf::is_fixed_width<ElementTo>(), "Unexpected non-fixed width type.");
-  auto transformer     = fixed_width_type_converter<ElementFrom, ElementTo>{};
-  auto transform_begin = thrust::make_transform_iterator(begin, transformer);
-  auto const size      = cudf::distance(begin, end);
-  auto const elements  = thrust::host_vector<ElementTo>(transform_begin, transform_begin + size);
-  return rmm::device_buffer{elements.data(), size * sizeof(ElementTo), rmm::cuda_stream_default};
-}
+template <typename From, typename To>
+struct to_rep {
+  template <typename FromT = From, typename ToT = To>
+  constexpr ToT operator()(FromT fp) const
+  {
+    return fp.value();
+  }
+};
```

Comment:
For some reason, replacing lambda with functor `to_rep` causes compilation error. I could not figure out the issue.
So I am reverting back to `if constexpr else` cases.

```
/usr/include/c++/9/type_traits:2784:11: error: no type named 'type' in 'struct std::invoke_result<cudf::test::detail::to_rep<numeric::fixed_point<int, numeric::Radix::BASE_10>, int>, int>'
 2784 |     using invoke_result_t = typename invoke_result<_Fn, _Args...>::type;
      |           ^~~~~~~~~~~~~~~



/home/karthikeyan/dev/rapids/cudf/cpp/include/cudf_test/column_wrapper.hpp:211:75: error: no match for 'operator+' (operand types are 'thrust::transform_iterator<cudf::test::detail::to_rep<numeric::fixed_point<int, numeric::Radix::BASE_10>, int>, const int*, thrust::use_default, thrust::use_default>' and 'const int')
  211 |     return make_device_elements<RepType>(transform_begin, transform_begin + size);

```